lib/DiscoveryStreamFeed.jsm

```diff
@@ -279,6 +250,12 @@ this.DiscoveryStreamFeed = class DiscoveryStreamFeed {
           await this.enable();
         }
         break;
+      case at.SYSTEM_TICK:
+        // Only refresh if we loaded once in .enable()
+        if (this.config.enabled && this.loaded) {
+          await this.refreshAll({updateOpenTabs: false});
```

Comment:
I think this'll end up firing update events on every system tick.

If I'm reading it correctly, it goes into a layout/component/spoc load function, gets cache. If it needs to get fresh data, it does. Then it sends the fresh data or cache to the reducer.

The old code was slightly less dry, but also only fired system tick updates if cache has expired.